src/linter.rs

```diff
@@ -141,10 +149,24 @@ impl Linter {
     let start = Instant::now();
 
     let mut filtered_diagnostics = context.check_ignore_directive_usage();
+    // Run `ban-unknown-rule-code`
+    let unknown_used_diagnostics = context.ban_unknown_rule_code();
+
+    if !unknown_used_diagnostics.is_empty() {
+      context.check_unknown_rule_code_used();
+    }
+
+    if self.check_unknown_rules
+      && !context
+        .file_ignore_directive()
+        .map(|f| f.has_code(ban_unknown_rule_code::CODE))
+        .unwrap_or(false)
+    {
+      filtered_diagnostics.extend(unknown_used_diagnostics);
+    }
+
```

Comment:
Can we move all the logic associated with `ban-unknown-rule-code` into `Context::ban_unknown_rule_code`?
I mean, it should be better if `Context::ban_unknown_rule_code` also processes the flag on whether the ignore directive is used or not, and just returns diagnostics if any. In `filter_diagnostics`, we would just do:

```rs
filtered_diagnostics.extend(context.ban_unknown_rule_code());
```

like `ban-used-ignore` does.